### PR TITLE
Moves SSH tests to testify/testing package

### PR DESCRIPTION
**Pull request by tcsc**
Refactors the SSH tests to use the standard library `testing` package.

I couldn't make the existing GoCheck test suite do what I needed when writing a test for #6091. Not, at least, without duplicating a heap of code and breaking a lot of stuff. So rather than spend ages trying to inject the flexibility I needed into a deprecated test library, I used that time to remove it entirely.

The main changes in the PR are:
 * Removal of the `Check` test suite. All tests in the suite are now simple, standalone tests (which, incidentally, makes them _heaps_ easier to run in a debugger as I found out :smiley:).
 * The test setup provided by `GoCheck` has been replaced by a fixture that can be created & customised on a per-test basis.
 * The test teardown provided by `GoCheck` has been replaced by the `testing` package's cleanup functions
 * Test setup that is only used in a few tests (e.g. the `proxyClient`) has been moved from the fixture an into independent functions so as not to complicate the tests that don't need it.
 * Test fixture names have been updated in an attempt to be less ambiguous (e.g. you don't have to just _remember_ which one of `svr` and `sever` is the SSH server and which is the auth server any more) 
 * Asynchronous errors in the  X11 tests are now marshaled back into the main test goroutine for assertion.
 * Parallelises test execution

### CODE REVIEW

File: lib/srv/regular/sshserver_test.go [hunk 1]
```diff
@@ -104,77 +81,72 @@ func TestMain(m *testing.M) {
...
-	c.Assert(err, IsNil)
+	}
+	mutateCfg(&serverCfg)
 
-	// set up SSH client using the user private key for signing
-	up, err := s.newUpack(s.user, []string{s.user}, wildcardAllow)
-	c.Assert(err, IsNil)
+	testServer, err := auth.NewTestServer(serverCfg)
+	require.NoError(t, err)
+	t.Cleanup(func() { testServer.Shutdown(context.Background()) })
```
Comment (nklaassen) [inline lib/srv/regular/sshserver_test.go:121]:
```suggestion
	t.Cleanup(func() { require.NoError(t, testServer.Shutdown(context.Background())) })
```

File: lib/srv/regular/sshserver_test.go [hunk 2]
```diff
@@ -185,56 +157,112 @@ func (s *SrvSuite) SetUpTest(c *C) {
...
+		[]ssh.Signer{signer},
+		nodeClient,
+		nodeDir,
+		"",
+		utils.NetAddr{},
+		serverOptions...)
+	require.NoError(t, err)
+	require.NoError(t, auth.CreateUploaderDir(nodeDir))
+	require.NoError(t, sshSrv.Start())
+	t.Cleanup(func() { sshSrv.Close() })
```
Comment (nklaassen) [inline lib/srv/regular/sshserver_test.go:177]:
```suggestion
	t.Cleanup(func() { require.NoError(t, sshSrv.Close()) })
```

File: lib/srv/regular/sshserver_test.go [hunk 3]
```diff
@@ -185,56 +157,112 @@ func (s *SrvSuite) SetUpTest(c *C) {
...
+		HostKeyCallback: ssh.FixedHostKey(signer.PublicKey()),
 	}
 
-	client, err := ssh.Dial("tcp", s.srv.Addr(), s.cltConfig)
-	c.Assert(err, IsNil)
-	c.Assert(agent.ForwardToAgent(client, keyring), IsNil)
-	s.clt = client
+	client, err := ssh.Dial("tcp", sshSrv.Addr(), cltConfig)
+	require.NoError(t, err)
+	t.Cleanup(func() { client.Close() })
```
Comment (nklaassen) [inline lib/srv/regular/sshserver_test.go:206]:
```suggestion
	t.Cleanup(func() { require.NoError(t, client.Close()) })
```

**Review by nklaassen — APPROVED** [review]
I like this

**Review by Joerger — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Pull request merged**